odoo/tests/common.py

```diff
@@ -1569,6 +1569,23 @@ def get_http_request_threads():
             self._logger.info('remaining requests')
             odoo.tools.misc.dumpstacks()
 
+    def login_simple(self, username, password, disable_post_raise=False):
+        """Log in with provided credential and returns response to POST request."""
+        res_get = self.url_open('/web/login')
+        res_get.raise_for_status()
+        csrf_anchor = '<input type="hidden" name="csrf_token" value="'
+        csrf_token = res_get.text.partition(csrf_anchor)[2].partition('"')[0]
```

Comment:
maybe instead of this, can you use this ?
```python
Request.csrf_token(self)
```

(I know it's not you, but we can clean it I think)

Reply:
I don't have easy access to the request from here. I'll keep looking and ping you back once I know for sure ;)

Reply:
can't you just import request ?

Reply:
Hum indeed, the session id doesn't exist yet before you call `/web/login` for the first time :thinking: 

But maybe we can copy the behavior of `authenticate` and avoid useless HTTP request, what do you think ?

```python
def login_simple(self, username, password):
    """Log in with provided credential and returns response to POST request."""
    self.session = session = odoo.http.root.session_store.new()
    self.opener = Opener(self.cr)
    self.opener.cookies['session_id'] = session.sid

    res_post = self.url_open('/web/login', data={
        'login': username,
        'password': password,
        'csrf_token': http.Request.csrf_token(self),
    }, headers={'Set-Cookie': 'session_id=' + session.sid})

    res_post.raise_for_status()

    return res_post
```